internal/pkg/agent/cmd/install.go

```diff
@@ -42,7 +42,9 @@ would like the Agent to operate.
 
 	cmd.Flags().BoolP("force", "f", false, "Force overwrite the current and do not prompt for confirmation")
 	cmd.Flags().BoolP("non-interactive", "n", false, "Install Elastic Agent in non-interactive mode which will not prompt on missing parameters but fails instead.")
-	cmd.Flags().String(flagInstallBasePath, paths.DefaultBasePath, "The path where the Elastic Agent will be installed. It must be an absolute path.")
+	// We are not supporting a custom base path, supplied via the `--base-path` CLI flag, just yet because
+	// we don't have Endpoint support for it yet. See also: https://github.com/elastic/elastic-agent/pull/2592
+	//cmd.Flags().String(flagInstallBasePath, paths.DefaultBasePath, "The path where the Elastic Agent will be installed. It must be an absolute path.")
```

Comment:
Instead of preventing it from being set, should we just mark the flag as hidden, with `MarkHidden("--base-path")`? That would allow experimental usage of it. Just a thought.

Reply:
I'll defer to @cmacknz on this, but I think we really want to ensure that this flag cannot be used at all.  But if that's not the case, then I'm ++ for the hidden idea.

Reply:
If someone uses a hidden and undocumented feature, it would be an unsupported use case.

I'm ok with making this hidden if it makes it easier for all of the tests we've written around it to keep working (the integration tests for example). We are going to keep this feature in some form, I would prioritize preventing code rot.

Reply:
Sounds good, thanks @blakerouse and @cmacknz.  Will make this a hidden flag.

Reply:
Done.  So much neater!